src/Razor/test/Microsoft.AspNetCore.Razor.Language.Test/DefaultRazorTagHelperBinderPhaseTest.cs

```diff
@@ -1342,6 +1342,27 @@ public void IsTagHelperFromMangledClass_WorksAsExpected()
             Assert.True(result);
         }
 
+        [Theory]
+        [InlineData("", false, "", "")]
+        [InlineData(".", true, "", "")]
+        [InlineData("Foo", true, "", "Foo")]
+        [InlineData("SomeProject.Foo", true, "SomeProject", "Foo")]
+        [InlineData("SomeProject.Foo<Bar>", true, "SomeProject", "Foo<Bar>")]
+        [InlineData("SomeProject.Foo<Bar.Baz>", true, "SomeProject", "Foo<Bar.Baz>")]
+        [InlineData("SomeProject.Foo<Bar.Baz>>", true, "", "SomeProject.Foo<Bar.Baz>>")]
+        [InlineData("SomeProject..Foo<Bar>", true, "SomeProject.", "Foo<Bar>")]
+        public void TrySplitNamespaceAndType_WorksAsExpected(string fullTypeName, bool expectedResult, string expectedNamespace, string expectedTypeName)
```

Comment:
What about namespaces that contain a `.` - would it be worth added more `InlineData` to cover these scenarios?

E.g.
```
        [InlineData("SomeProject.SomeNamespace.Foo", true, "SomeProject.SomeNamespace", "Foo")]
        [InlineData("SomeProject.SomeNamespace.Foo<Bar>", true, "SomeProject.SomeNamespace", "Foo<Bar>")]
        [InlineData("SomeProject.SomeNamespace.Foo<Bar.Baz>", true, "SomeProject.SomeNamespace", "Foo<Bar.Baz>")]
        [InlineData("SomeProject.SomeNamespace.Foo<Bar.Baz>>", true, "", "SomeProject.SomeNamespace.Foo<Bar.Baz>>")]
        [InlineData("SomeProject.SomeNamespace..Foo<Bar>", true, "SomeProject.SomeNamespace.", "Foo<Bar>")]
```

Reply:
@adrianwright109, that probably won't add value as we stop and return whatever is left once we parse the type name.